### ISSUE TITLE
Rule proposal: `prefer-object-defineProperties`

### ISSUE THREAD

**Issue by Pyrolistical** [sindresorhus/eslint-plugin-unicorn#1729]
### Description

I didn't know [`Object.defineProperties`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperties) existed. Would have been nice for a rule to teach me as [I wrote consecutive `Object.defineProperty`](https://github.com/concept-not-found/module-linking-polyfill/blob/b7bc16056f4faf42173cc4fd334d4ef177df0121/packages/transformer/parser/grammar.js#L76-L83).

### Fail

```js
Object.defineProperty(matcher, 'logger', {
  value: () => {},
  writable: true,
})
Object.defineProperty(matcher, 'builder', {
  value: (value) => value.map(({ build }) => build()),
  writable: true,
})
```

### Pass

```js
Object.defineProperties(matcher, {
  logger: {
    value: () => {},
    writable: true,
  },
  builder: {
    value: (value) => value.map(({ build }) => build()),
    writable: true,
  }
})
```

**Comment by fisker** [comment]
If there are many descriptors, normally it's a loop, if there are not many descriptors, I don't think they make much difference.

**Comment by fisker** [comment]
If we decide to add this rule, we should also check [`Reflect.defineProperty`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/defineProperty)

**Comment by Pyrolistical** [comment]
Another reason to reject this rule, it would interact poorly with https://github.com/microsoft/TypeScript/issues/41424

**Comment by sindresorhus** [comment]
> Another reason to reject this rule, it would interact poorly with https://github.com/microsoft/TypeScript/issues/41424

This plugin targets JavaScript. TypeScript weaknesses is not a concern. TypeScript users can just disable this rule.

**Comment by sindresorhus** [comment]
I think if this is going to be useful, it needs an auto-fix. Manually converting to `Object.defineProperties` would not be worth the effort.

**Comment by sindresorhus** [comment]
Accepted
